### PR TITLE
flyte-cli setup-config throws JSONDecodeError

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
flyte-cli setup-config throws JSONDecodeError because we tried to get admin credential in insecure mode. 

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Get credentials only in a secure mode.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/163

## Follow-up issue
_NA_

image

### CODE REVIEW

File: flytekit/clis/flyte_cli/main.py [hunk 1]
```diff
@@ -516,13 +499,51 @@ def make_context(self, cmd_name, args, parent=None):
...
         # This is where we handle the value read from the flyte-cli config file, if any, for the insecure flag.
         # Previously we tried putting it into the default into the declaration of the option itself, but in click, it
         # appears that flags operate like toggles. If both the default is true and the flag is passed in the command,
         # they negate each other and it's as if it's not passed. Here we rectify that.
         if _INSECURE_FLAG and _INSECURE_FLAGS[0] not in prefix_args:
             prefix_args.append(_INSECURE_FLAGS[0])
 
+        # Use host url in config file if users don't specify the host url
+        if _HOST_FLAGS[0] not in prefix_args:
+            prefix_args.extend([_HOST_FLAGS[0], _six.text_type(_HOST_URL)])
```
Comment (wild-endeavor) [inline flytekit/clis/flyte_cli/main.py:546]:
```suggestion
            prefix_args.extend([_HOST_FLAGS[0], str(_HOST_URL)])
```

**Comment by wild-endeavor** [comment]
I installed the sha from this PR into a new virtualenv and then ran the following.

### host location
```
$ flyte-cli -h localhost:30081 -i setup-config
Config file not found at default location, relying on environment variables instead.
                        To setup your config file run 'flyte-cli setup-config'
Usage: flyte-cli setup-config [OPTIONS]
Try 'flyte-cli setup-config --help' for help.

Error: Missing option '-h' / '--host'.
```
Is this intended?  Are you supposed to have to specify the host after `setup-config`?


### client_id missing
```
$ flyte-cli setup-config -h localhost:30081 -i
Config file not found at default location, relying on environment variables instead.
                        To setup your config file run 'flyte-cli setup-config'
Welcome to Flyte CLI! Version: 0.0.0+develop
Traceback (most recent call last):
  File "/Users/ytong/envs/tstsetup/bin/flyte-cli", line 8, in <module>
    sys.exit(_flyte_cli())
  File "/Users/ytong/envs/tstsetup/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/ytong/envs/tstsetup/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/ytong/envs/tstsetup/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/ytong/envs/tstsetup/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/ytong/envs/tstsetup/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/ytong/envs/tstsetup/lib/python3.8/site-packages/flytekit/clis/flyte_cli/main.py", line 2372, in setup_config
    f.write("client_id={}".format(data["client_id"]))
KeyError: 'client_id'
```
Looking at the response, it looks like Admin is returning:
`{'error': 'unknown service flyteidl.service.AuthMetadataService', 'code': 12, 'message': 'unknown service flyteidl.service.AuthMetadataService'}` when auth isn't on.

The file is still being written correctly:
```
$ cat ~/.flyte/config
[platform]
url = localhost:30081
insecure = True
```
but the stack trace is ugly and made me think the entire command failed.  Can we hide it somehow?

File: flytekit/clis/flyte_cli/main.py [hunk 2]
```diff
@@ -2359,26 +2367,27 @@ def setup_config(host, insecure):
...
-        f.write("[platform]")
-        f.write("\n")
-        f.write("url={}".format(host))
-        f.write("\n")
-        f.write("insecure={}".format(insecure))
-        f.write("\n\n")
-
-        f.write("[credentials]")
-        f.write("\n")
-        f.write("client_id={}".format(data["client_id"]))
```
Comment (pingsutw) [inline flytekit/clis/flyte_cli/main.py:2372]: cc @wild-endeavor. 
```
File "/Users/ytong/envs/tstsetup/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/ytong/envs/tstsetup/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/ytong/envs/tstsetup/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/ytong/envs/tstsetup/lib/python3.8/site-packages/flytekit/clis/flyte_cli/main.py", line 2372, in setup_config
    f.write("client_id={}".format(data["client_id"]))
KeyError: 'client_id'
```
It seems like you used the old version of flytekit.
I already remove `f.write`

**Comment by pingsutw** [comment]
> Is this intended?

I didn't test it before.

I just ran the command `flyte-cli -h localhost:30081 -i setup-config`  in my local laptop, and it throws an error. Will fix it ASAP.
Error messages:
```
    response = _requests.get(config_url)
  File "/Users/kevin/opt/anaconda3/envs/flyte/lib/python3.8/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/Users/kevin/opt/anaconda3/envs/flyte/lib/python3.8/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Users/kevin/opt/anaconda3/envs/flyte/lib/python3.8/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/kevin/opt/anaconda3/envs/flyte/lib/python3.8/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/Users/kevin/opt/anaconda3/envs/flyte/lib/python3.8/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='none', port=443): Max retries exceeded with url: /config/v1/flyte_client (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f82b0d91220>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))
```

**Comment by pingsutw** [comment]
Fixed it.

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**